app/components/myLibrary/libraryView.js

```diff
@@ -33,23 +28,46 @@ export default class LibraryView extends Component {
     this.state = {
       dataSource: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
       onPressedBookIndex: undefined,
-      onPressedColumn: undefined,
-      initialBookColor: [],
       fadeAnimRow: new Animated.Value(0),
       offset: 0,
-      direction: undefined
+      direction: undefined,
+      pan: new Animated.ValueXY()
     };
-    for (let i = 0; i < this.state.dataSource.length; i = i + 1) {
-      this.state.initialBookColor.push(generateRandomColor());
-    }
+    this.panResponder = PanResponder.create({
+      onStartShouldSetPanResponder: () => {console.log('touched');},
+      onPanResponderMove: Animated.event([null, {
+        dx: this.state.pan.x,
+        dy: this.state.pan.y
+      }]),
+      onPanResponderRelease : (e, gesture) => {}
+    });
+    this.renderAbsoluteRow = this.renderAbsoluteRow.bind(this);
   }
 
   static propTypes = {
     user: PropTypes.object,
     libraryList: PropTypes.any,
-    prevScene: PropTypes.string
+    prevScene: PropTypes.string,
+    coordinate: PropTypes.array,
+    initialBookColor: PropTypes.array,
+    sortRows: PropTypes.func
   };
 
+  shouldComponentUpdate(nextState) {
```

Comment:
this is in order to optimize performance. Here is the ref FYI: https://facebook.github.io/react/docs/optimizing-performance.html#shouldcomponentupdate-in-action